Library/Homebrew/brew.sh

```diff
@@ -402,15 +409,17 @@ Minimum required version: ${HOMEBREW_MINIMUM_CURL_VERSION}
 Your curl version: ${curl_name_and_version##* }
 Your curl executable: $(type -p $HOMEBREW_CURL)"
 
-    if [[ -z $HOMEBREW_CURL_PATH || -z $HOMEBREW_DEVELOPER ]]; then
+    if [[ -n $HOMEBREW_CURL_PATH && -z $HOMEBREW_DEVELOPER ]]
+    then
+      odie "$message"
+    else
       HOMEBREW_SYSTEM_CURL_TOO_OLD=1
       HOMEBREW_FORCE_BREWED_CURL=1
-      if [[ -z $HOMEBREW_CURL_WARNING ]]; then
+      if [[ -z $HOMEBREW_CURL_WARNING ]]
+      then
         onoe "$message"
         HOMEBREW_CURL_WARNING=1
       fi
-    else
-      odie "$message"
```

Comment:
I actually think the only situation where we should print a warning any more is if the `HOMEBREW_CURL_PATH` `curl` is too old as:
- if the system `curl` is too old: we will set `HOMEBREW_SYSTEM_CURL_TOO_OLD` and `HOMEBREW_FORCE_BREWED_CURL` and proceed accordingly
- if they have already set `HOMEBREW_FORCE_BREWED_CURL`: we don't want to print a warning/error but instead proceed to try and install it (in `update.sh` which may fail, at which point we should tell them about `HOMEBREW_CURL_PATH`)

Thoughts?

Reply:
What parts of `brew` require the specified `git` version? Is the installation of brewed `git` guaranteed to work?

Could be worded as a warning with the extra "Brewed git will now be installed" or something, and then silence the warning on future runs if brewed curl is detected as installed.

Reply:
> What parts of `brew` require the specified `curl`/`git` version?

- any part that uses `curl` to download anything using HTTPS (which ends up being: almost everything including even Bintray on some old versions)
- any part that uses `git` to connect to GitHub

> Is the installation of brewed `curl`/`git` guaranteed to work?

No but it it fails it'll then bail on `brew update` entirely. We should/could add a check to ensure that update is run immediately if `HOMEBREW_FORCE_BREWED_CURL` isn't set and refuse to run if `HOMEBREW_NO_AUTO_UPDATE` is set.

> Could be worded as a warning with the extra "Brewed curl/git will now be installed" or something, and then silence the warning on future runs if brewed `curl`/`git` is detected as installed.

Yes, something like this seems appropriate but perhaps better placed to be when the installation is about to happen e.g. in `update.sh`

Reply:
The point I'm making is that if it does fail to install, it should be made clear that the solution is updating system `curl`/`git`.

Reply:
Yeh, agreed 👍🏻 and/or setting this variable.